File: Year-2023/LeetCode-September-challenge/MonotonicArray.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        boolean increase = false, decrease = false;

        for(int i = 1; i < nums.length; ++i){
            if(nums[i] > nums[i - 1]) increase = true;
            else if(nums[i] < nums[i - 1]) decrease = true;
            if(increase && decrease) return false;
        }
        return true;
    }
}
public class MonotonicArray{
    public static void main(String[] args) {
        int[] nums = {1,2,2,3};
        System.out.println(new Solution().isMonotonic(nums));
    }
}

File: Year-2023/LeetCode-September-challenge/MonotonicArray.js
var isMonotonic = function (nums) {
    let increase = false, decrease = false;

    for (let i = 1; i < nums.length; ++i) {
        if (nums[i] > nums[i - 1]) increase = true;
        else if (nums[i] < nums[i - 1]) decrease = true;
        if (increase && decrease) return false;
    }
    return true;
};

let nums = [1,2,2,3];
console.log(isMonotonic(nums));